cpp/tests/reductions/reduction_tests.cpp

```diff
@@ -119,6 +116,34 @@ struct ReductionTest : public cudf::test::BaseFixture {
       EXPECT_ANY_THROW(statement());
     }
   }
+
+  // Test with initial value
+  template <typename T_out>
+  void reduction_test(cudf::column_view const& underlying_column,
+                      cudf::scalar const& initial_value,
+                      T_out expected_value,
+                      bool succeeded_condition,
+                      std::unique_ptr<reduce_aggregation> const& agg,
+                      cudf::data_type output_dtype = cudf::data_type{},
+                      bool expected_null           = false)
+  {
+    if (cudf::data_type{} == output_dtype) output_dtype = underlying_column.type();
+
+    auto statement = [&]() {
+      std::unique_ptr<cudf::scalar> result =
+        cudf::reduce(underlying_column, agg, output_dtype, initial_value);
+      using ScalarType = cudf::scalar_type_t<T_out>;
+      auto result1     = static_cast<ScalarType*>(result.get());
+      EXPECT_EQ(expected_null, !result1->is_valid());
+      if (result1->is_valid()) { EXPECT_EQ(expected_value, T_out{result1->value()}); }
```

Comment:
Do not put the test macro in functions. This is important! When a test failed, the macro will tell you which line failed. If the macro is here, you will have no way to know which test has failed.

The utility functions like this should return the tested return (and the expected result, if necessary). The function return type maybe `std::pair<std::unique_ptr<column>, std::unique_ptr<column>>` or something similar. Then, at the caller site, call the `EXPECT_EQ` macro.